### PR TITLE
fix: add typescript declarations

### LINKED ISSUE #37: TypeScript: Deceleration or Migrate
**What pain point are you perceiving?.**
Even when including `@types/marked`, including `marked-mangle` doesn't support TypeScript.

**Describe the solution you'd like**
[As Marked migrates to TypeScript](https://github.com/markedjs/marked/pull/2805), so should these new Marked support libraries. As they are small projects and the main project will remove the `@types/marked` dependency, I would request the migration route to reduce the need to add two packages per package.

### PR DESCRIPTION
**Pull request by mrsimonemms**
Add TypeScript declarations.

This follows the principles of https://github.com/markedjs/marked-gfm-heading-id/pull/324

Fixes #37

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import type { marked } from "marked"
+
+/**
+ * Mangle email address
+ *
+ * @param options Options for the extension
```
Comment (UziTech) [inline src/index.d.ts:6]: Can you remove this param since there are no options?

**Review by UziTech — CHANGES_REQUESTED** [review]

File: src/index.d.ts (hunk 1 above)
Comment (mrsimonemms) [reply src/index.d.ts]: Done. Well spotted

Comment (mrsimonemms) [reply src/index.d.ts]: Done. Well spotted

**Review by UziTech — APPROVED** [review]
Thanks! :100:

**Pull request merged**